src/ZoneServer/World/Actors/CombatEntities/Components/MovementComponent.cs

```diff
@@ -404,12 +404,21 @@ private void CheckWarp()
 		/// <param name="type"></param>
 		public void SetMoveSpeedType(MoveSpeedType type)
 		{
-			if (this.Entity is Mob mob && this.MoveSpeedType != type)
+			if (this.MoveSpeedType == type)
+				return;
+
+			if (this.Entity is Mob mob)
 			{
 				this.MoveSpeedType = type;
 				this.Entity.Properties.Invalidate(PropertyName.MSPD);
 
 				Send.ZC_MSPD(this.Entity);
+			} else if (this.Entity is Character character && character.IsDummy)
+			{
+				this.MoveSpeedType = type;
+				character.Properties.Invalidate(PropertyName.MSPD);
+
+				Send.ZC_MSPD(character.Owner, character);
```

Comment:
Can you  tell me what the exact purpose of this `else if` is? It seems like you could just pass the character into `ZC_MSPD`, which would allow you to leave it as one if.

Reply:
This is needed because DummyCharacters don't have connection so I need to pass the Owner to the other method.